src/index.ts

```diff
@@ -0,0 +1,346 @@
+      Record<'Language' | 'Role' | 'Country' | 'Cookie', readonly string[]>
+    >;
+  }>;
+  headers: Array<{
+    for: string;
+    values: unknown; // marked as unknown because is not required here.
+  }>;
+  functions: {
+    directory: string;
+  };
+  build: {
+    command: string;
+    environment: Record<string, string>;
+    edge_functions: string;
+    processing: Record<string, unknown>;
+  };
+};
+type Constants = {
+  CONFIG_PATH?: string;
+  PUBLISH_DIR: string;
+  FUNCTIONS_SRC: string;
+  FUNCTIONS_DIST: string;
+  IS_LOCAL: boolean;
+  NETLIFY_BUILD_VERSION: `${string}.${string}.${string}`;
+  SITE_ID: string;
+};
+
+/**
+ * this type is built based on the content of the plugin manifest file
+ * Information found here https://docs.netlify.com/integrations/build-plugins/create-plugins/#inputs
+ */
+type Inputs = {
+  cloudName: string;
+  deliveryType: string;
+  imagesPath: string;
+  folder: string;
+  uploadPreset: string;
+  loadingStrategy: string;
+};
+
+type Utils = {
+  build: {
+    failBuild: (message: string, { error }?: { error: Error }) => void;
+    failPlugin: (message: string, { error }?: { error: Error }) => void;
+    cancelBuild: (message: string, { error }?: { error: Error }) => void;
+  };
+  status: {
+    show: ({
+      title,
+      summary,
+      text,
+    }: {
+      title: string;
+      summary: string;
+      text: string;
+    }) => void;
+  };
+};
+type OnBuildParams = {
+  netlifyConfig: NetlifyConfig;
+  constants: Constants;
+  inputs: Inputs;
+  utils: Utils;
+};
+type OnPostBuildParams = Omit<OnBuildParams, 'netlifyConfig'>;
+
+const CLOUDINARY_ASSET_DIRECTORIES = [
+  {
+    name: 'images',
+    inputKey: 'imagesPath',
+    path: '/images',
+  },
+];
+
+/**
+ * TODO
+ * - Handle srcset
+ */
+
+const _cloudinaryAssets = { images: {} } as Assets;
+
+export async function onBuild({
+  netlifyConfig,
+  constants,
+  inputs,
+  utils,
+}: OnBuildParams) {
+  console.log('[Cloudinary] Creating redirects...');
+
+  const isProduction = process.env.CONTEXT === 'production';
+  const host = isProduction
+    ? process.env.NETLIFY_HOST
+    : process.env.DEPLOY_PRIME_URL;
+
+  console.log(`[Cloudinary] Using host: ${host}`);
+
+  const { PUBLISH_DIR } = constants;
+
+  const {
+    deliveryType,
+    uploadPreset,
+    folder = process.env.SITE_NAME,
+    imagesPath = CLOUDINARY_ASSET_DIRECTORIES.find(
+      ({ inputKey }) => inputKey === 'imagesPath',
+    )?.path,
+  } = inputs;
+
+  if (!folder) {
+    utils.build.failPlugin(ERROR_SITE_NAME_REQUIRED);
+    return;
+  }
+
+  if (!host && deliveryType === 'fetch') {
+    console.warn(`[Cloudinary] ${ERROR_NETLIFY_HOST_UNKNOWN}`);
+    console.log(`[Cloudinary] ${ERROR_NETLIFY_HOST_CLI_SUPPORT}`);
+    return;
+  }
+
+  const cloudName = process.env.CLOUDINARY_CLOUD_NAME || inputs.cloudName;
+  const apiKey = process.env.CLOUDINARY_API_KEY;
+  const apiSecret = process.env.CLOUDINARY_API_SECRET;
+
+  if (!cloudName || !apiKey || !apiSecret) {
```

Comment:
@matiasfha was there a context to the addition of `!apiKey` and `!apiSecret` here? this now breaks builds that are only using the fetch API, which only requires cloud name

the plkugin hasn't been updated on netlify so its not impacting anyone at the moment but will once updated

before fixing this curious why this was added

Reply:
IIRC this was added to satisfy TS
Let me create another PR to remove that restriction and also add a test for that build case

Reply:
This was the TS error
<img width="691" alt="CleanShot 2023-09-19 at 10 49 35@2x" src="https://github.com/cloudinary-community/netlify-plugin-cloudinary/assets/282006/f725a754-75d3-4d32-9443-02943adb1e2d">

I'm working on  a change